<launch>

  <arg name="image_view" default="true" />

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />

    <param name="intrinsic_controls/focus_automatic_continuous" value="false" />
    <param name="image_width" value="1920" />
    <param name="image_height" value="1080" />
    <param name="pixel_format" value="yuv" />
    <param name="color_format" value="yuv422p" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="camera_info_url" value="file://$(find hmip_framework)/config/calibration/usb_camera_calibration_1920.yaml"/>    
  </node>

  <!-- TRACKERS -->
  <node name="gesture_tracker" pkg="hand_gaze_trackers" type="gesturetrack.py" output="screen" >

     <rosparam command="load" file="$(find hmip_framework)/config/object_locations_gestures.yaml" />
     <param name="mindistance" value="0.05"/>
  </node>


  <node if="$(arg image_view)" name="image_view" pkg="image_view" type="image_view"
        respawn="false" output="screen">
    <remap from="image" to="/hand_tracker/image"/>
    <param name="autosize" value="true" />
  </node>
  
</launch>
